pkg/api/v1/types.go

```diff
@@ -2165,6 +2165,14 @@ type PodProxyOptions struct {
 	Path string `json:"path,omitempty"`
 }
 
+// ServiceProxyOptions is the query options to a Service's proxy call.
+type ServiceProxyOptions struct {
```

Comment:
@lavalamp @liggitt  Should I combine/reuse PodProxyOptions in this PR, or  could I open a new PR to combine/resue? Do we need to rename PodProxyOptions to SubresourceProxyOptions? Looking forward to your reply.